common/src/chain/block/consensus_data.rs

```diff
@@ -50,30 +50,43 @@ impl ConsensusData {
     }
 }
 
-/// Fake PoS just to test spending block rewards; will be removed at some point in the future
+/// Data required to validate a block according to the PoS consensus rules.
 #[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
 pub struct PoSData {
+    /// Inputs for block reward
     kernel_inputs: Vec<TxInput>,
     kernel_witness: Vec<InputWitness>,
+
+    /// Id of the stake pool used for target calculations
     stake_pool_id: PoolId,
-    vrf_data: VRFReturn,
-    bits: Compact,
+
+    /// VRF data used for calculating hash below the target.
+    /// It represents random seed generated based on the randomness of the sealed epoch.
+    vrf_data_from_sealed_epoch: VRFReturn,
+
+    /// VRF data used for calculating randomness for the current block.
+    /// It represents random seed generated based on the randomness of the previous block.
+    vrf_data_from_prev_block: VRFReturn,
```

Comment:
This is wrong. There should be only one vrf_data.

Reply:
I don't see how it's possible with only one vrf data. It is used in 2 use cases: sealed epoch randomness + vrf data to mine a block; prev block randomness + vrf data to create a randomness for the new block which will become sealed randomness at some point.

Reply:
done, from now on a single vrf is used. I reconsidered the code and removed `PreconnectData` and related logic. Every block generates randomness based on a sealed epoch. And the randomness of the last block is used as a randomness of an epoch.